logos_setup.py

```diff
@@ -0,0 +1,801 @@
+    FLPRODUCT = os.environ.get('FLPRODUCT')
+    FLPRODUCTi = os.environ.get('FLPRODUCTi')
+    LOGOS_ICON_FILENAME = os.environ.get('LOGOS_ICON_FILENAME')
+
+    if not os.environ.get("LOGOS_ICON_URL"):
+        os.environ["LOGOS_ICON_URL"] = "https://raw.githubusercontent.com/ferion11/LogosLinuxInstaller/master/img/" + os.environ.get("FLPRODUCTi") + "-128-icon.png"
+
+    logos_icon_path = os.path.join(APPDIR, "data", LOGOS_ICON_FILENAME)
+
+    if not os.path.isfile(logos_icon_path):
+        os.makedirs(os.path.join(APPDIR, "data"), exist_ok=True)
+        raw_bytes = None
+        try:
+            with urllib.request.urlopen(LOGOS_ICON_URL) as f:
+                raw_bytes = f.read()
+        except urllib.error.URLError as e:
+            logos_error(e)
+        if raw_bytes is not None:
+            with open(logos_icon_path, 'wb') as f:
+                f.write(raw_bytes)
+
+    desktop_entry_path = os.path.expanduser(f"~/.local/share/applications/{FLPRODUCT}Bible.desktop")
+    if os.path.exists(desktop_entry_path):
+        os.remove(desktop_entry_path)
+
+    with open(desktop_entry_path, 'w') as desktop_file:
+        desktop_file.write(f"""[Desktop Entry]
+Name={FLPRODUCT}Bible
+Comment=A Bible Study Library with Built-In Tools
+Exec={APPDIR}/Logos.sh
+Icon={logos_icon_path}
+Terminal=false
+Type=Application
+Categories=Education;
+""")
+
+    os.chmod(desktop_entry_path, 0o755)
+
+def createLauncher():
+    '''
+	getScriptTemplate "Launcher-Template.sh";
+	printf "%s\n" "$(apply_shell_expansion "${WORKDIR}/Launcher-Template.sh")" > "${WORKDIR}"/"${FLPRODUCT}".sh;
+	chmod 755 "${WORKDIR}"/"${FLPRODUCT}".sh;
+	mv "${WORKDIR}"/"${FLPRODUCT}".sh "${INSTALLDIR}"/
+    '''
+    # NOTE: This creates a basic place-holder script just to be able to launch
+    #   the installed app.
+    HOME = os.getenv('HOME')
+    CONFIG_PATH = os.getenv('CONFIG_PATH', f'{HOME}/.config/Logos_on_Linux/Logos_on_Linux.conf')
+    FLPRODUCT = os.getenv('FLPRODUCT')
+    INSTALLDIR = os.getenv('INSTALLDIR')
+    LOGOS_EXE = os.getenv('LOGOS_EXE')
+    WINE_EXE = os.getenv('WINE_EXE')
+    WORKDIR = os.getenv('WORKDIR')
+    launcher_script = Path(f"{WORKDIR}/{FLPRODUCT}.sh")
+    launcher_script.write_text('#!/usr/bin/env bash\n')
+    with launcher_script.open('a') as f:
+        with open(CONFIG_PATH) as c:
+            for k, v in sorted(json.load(c).items()):
+                f.write(f'{k}="{v}"\nexport {k}\n')
+        f.write(f'\n"${{WINE_EXE}}" "${{LOGOS_EXE}}"\n')
+        f.write(f'"${{WINESERVER_EXE}}" -w\n')
+    launcher_script.chmod(0o755)
+    launcher_script.rename(f"{INSTALLDIR}/{launcher_script.name}")
+
+def createControlPanel():
+    '''
+	getScriptTemplate "controlPanel-Template.sh";
+	printf "%s\n" "$(apply_shell_expansion "${WORKDIR}/controlPanel-Template.sh")" > "${WORKDIR}"/controlPanel.sh;
+	chmod 755 "${WORKDIR}"/controlPanel.sh;
+	mv "${WORKDIR}"/controlPanel.sh "${INSTALLDIR}"/
+    '''
+    pass
```

Comment:
This can be deleted. No longer needed with python

Reply:
Done in [879e571](https://github.com/thw26/LogosLinuxInstaller/pull/7/commits/879e57183f09f4103d56270aed67f95eb77ef897)